coffeecard/CoffeeCard.Library/Services/v2/TicketService.cs

```diff
@@ -49,5 +55,46 @@ public Task<List<TicketResponse>> GetTickets(User user, bool includeUsed)
                     ProductName = t.Purchase.ProductName
                 }).ToListAsync();
         }
+        
+        public async Task<TicketDto> UseTicket(int userId, int productId)
+        {
+            Log.Information($"Using ticket with id, {productId}");
+            var ticket = GetFirstTicketFromProduct(productId, userId);
+
+            ticket.IsUsed = true;
+            ticket.DateUsed = DateTime.UtcNow;
+            
+            if (ticket.Purchase.Price != 0) //Paid products increases your rank on the leaderboard
+            {
+                await _statisticService.IncreaseStatisticsBy(userId, 1);
+            }
+
+            await _context.SaveChangesAsync();
+
+            return new TicketDto
```

Comment:
In API v2, we do not name classes DTO, instead, we specialize them for request/response, e.g. the name could be `UsedTicketResponse`, idk